catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -611,15 +624,21 @@ protected UpdateResponse patchMetacards(List<MetacardChanges> metacardChanges)
             for (AttributeChange attributeChange : changeset.getAttributes()) {
                 for (String id : changeset.getIds()) {
                     List<String> values = attributeChange.getValues();
-                    Metacard result = results.get(id)
-                            .getMetacard();
+
+                    Result result = results.get(id);
+                    if (result == null) {
+                        LOGGER.debug("Metacard with id {} not found in result set", id);
+                        continue;
+                    }
+
+                    Metacard metacard = result.getMetacard();
```

Comment:
I agree; It seems more reasonable to fail here with an exception than continue.  The main motivation for this was to prevent an observed potential NPE in the case that `results.get(id)` is null